app/src/main/java/com/pvp/app/ui/screen/task/GeneralTaskCreationForm.kt

```diff
@@ -0,0 +1,247 @@
+package com.pvp.app.ui.screen.task
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.DisplayMode
+import androidx.compose.material3.TextButton
+import com.pvp.app.model.Task
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+class GeneralTasksManager {
+
+    private val _tasks = mutableStateListOf<Task>()
+    val tasks: List<Task> get() = _tasks
```

Comment:
Redundant abstraction as in the end you are still publicly accessing whole `List` instance and can modify it outside as if it was public. Either use `val tasks = mutableStateListOf<Task>()` or make current `tasks.getter` return `_tasks.toList()` which would create a new immutable `List` instance upon getting every time. If chosen secondary option, you may have to adjust the associated code below.

However, tasks list showcase component (below) is not really necessary as only creation form is needed, which would persist data into the database and return user back to the calendar view where user could see the freshly created task card.